UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/ProductDataController.cs

```diff
@@ -95,12 +96,22 @@ public virtual async Task<IActionResult> PostProductIdentifiers([FromBody] strin
                     return BuildBadRequestErrorResponse(errors);
                 }
             }
+            AzureAdB2C azureAdB2C = new AzureAdB2C()
+            {
+                AudToken = TokenAudience,
+                IssToken = TokenIssuer
+            };
 
-            var productDetail = await productDataService.CreateProductDataByProductIdentifiers(productIdentifierRequest);
-
+            var productDetail = await productDataService.CreateProductDataByProductIdentifiers(productIdentifierRequest,azureAdB2C);
+            
+            if (productDetail.IsExchangeSetTooLarge)
+            {
+                Logger.LogInformation(EventIds.ExchangeSetTooLarge.ToEventId(), "Exchange Set is too large for Endpoint productIdentifiers");
```

Comment:
Can we change this to Logger.LogError. Also log co relation id and change message to "Requested exchange set is too large for product versions endpoint". Kindly do this for all 3 endpoints.